File: daily/Sliding Window/1248-Count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        return self.atMost(nums, k) - self.atMost(nums, k - 1)
    def atMost(self, nums: List[int], k: int) -> int:
        count, left, res = 0, 0, 0
        n = len(nums)
        for right in range(n):
            count += nums[right] % 2
            while count > k:
                count -= nums[left] % 2
                left += 1
            res += right - left + 1
        return res

File: daily/Sliding Window/1248-Count-number-of-nice-subarrays.go
package daily

func numberOfSubarrays(nums []int, k int) int {
	return atMost(nums, k) - atMost(nums, k-1)
}

func atMost(nums []int, k int) int {
	left, count, res := 0, 0, 0
	for right := range len(nums) {
		if nums[right]%2 != 0 {
			count += 1
		}
		for count > k {
			if nums[left]%2 != 0 {
				count -= 1
			}
			left++
		}
		res += right - left + 1
	}
	return res

}
